app/src/main/java/com/kodeco/android/countryinfo/ui/components/countryinfo/CountryInfoViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.kodeco.android.countryinfo.ui.components.countryinfo
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import com.kodeco.android.countryinfo.repositories.CountryRepository
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class CountryInfoViewModel(val repository: CountryRepository?) : ViewModel() {
```

Comment:
repository should be private. Pay attention to warnings in the IDE - they are your friend 😁